model/project_ref.go

```diff
@@ -420,6 +446,96 @@ func (p *ProjectRef) AddToRepoScope(user *user.DBUser) error {
 	return nil
 }
 
+// DetachFromRepo removes the branch from the relevant repo scopes, and updates the project to not point to the repo.
+// Any values that previously defaulted to repo will have the repo value explicitly set.
+func (p *ProjectRef) DetachFromRepo(u *user.DBUser) error {
+	before, err := GetProjectSettingsEventById(p.Id)
+	if err != nil {
+		return errors.Wrap(err, "error getting before project settings event")
+	}
+
+	// remove from relevant repo scopes
+	if err = p.RemoveFromRepoScope(); err != nil {
+		return err
+	}
+
+	mergedProject, err := FindMergedProjectRef(p.Id)
+	if err != nil {
+		return errors.Wrap(err, "error finding merged project ref")
+	}
+
+	// Save repo variables that don't exist in the repo as the project variables.
+	// Wait to save merged project until we've gotten the variables.
+	mergedVars, err := FindMergedProjectVars(before.ProjectRef.Id)
+	if err != nil {
+		return errors.Wrap(err, "error finding merged project vars")
+	}
+
+	mergedProject.UseRepoSettings = false
+	mergedProject.RepoRefId = ""
+	if err = mergedProject.Upsert(); err != nil {
+		return errors.Wrap(err, "error detaching project from repo")
+	}
+
+	// catch any resulting errors so that we log before returning
+	catcher := grip.NewBasicCatcher()
+	if mergedVars != nil {
+		_, err = mergedVars.Upsert()
+		catcher.Wrap(err, "error saving merged vars")
+	}
+
+	if len(before.Subscriptions) == 0 {
+		// Save repo subscriptions as project subscriptions if none exist
```

Comment:
Why do we only do this if none exist? Can we have a situation where a project relies on both repo level and project level subscriptions?

Reply:
nope; the project either defaults to repo subscriptions, or defines its own (this is only different for variables)